include/revng/Model/Types.h

```diff
@@ -0,0 +1,903 @@
+};
+INTROSPECTION_NS(flatc,
+                 FunctionPtrType,
+                 UUID,
+                 Name,
+                 Kind,
+                 ReturnType,
+                 ArgumentTypes,
+                 IsVariadic);
+
+// Make flatc::TypeKind yaml-serializable, required for making flatc::Type
+// yaml-serializable as well
+template<>
+struct llvm::yaml::ScalarEnumerationTraits<flatc::TypeKind::Values> {
+  static void enumeration(IO &io, flatc::TypeKind::Values &Val) {
+    io.enumCase(Val, "TypeKind::Invalid", flatc::TypeKind::Invalid);
+    io.enumCase(Val, "TypeKind::Primitive", flatc::TypeKind::Primitive);
+    io.enumCase(Val, "TypeKind::Enum", flatc::TypeKind::Enum);
+    io.enumCase(Val, "TypeKind::Typedef", flatc::TypeKind::Typedef);
+    io.enumCase(Val, "TypeKind::Struct", flatc::TypeKind::Struct);
+    io.enumCase(Val, "TypeKind::Union", flatc::TypeKind::Union);
+    io.enumCase(Val, "TypeKind::FunctionPtr", flatc::TypeKind::FunctionPtr);
+  }
+};
+
+// Make flatc::QualifierKind::Values yaml-serializable, required for making
+// flatc::Qualifier yaml-serializable as well
+template<>
+struct llvm::yaml::ScalarEnumerationTraits<flatc::QualifierKind::Values> {
+  static void enumeration(IO &io, flatc::QualifierKind::Values &Val) {
+    io.enumCase(Val, "QualifierKind::Invalid", flatc::QualifierKind::Invalid);
+    io.enumCase(Val, "QualifierKind::Pointer", flatc::QualifierKind::Pointer);
+    io.enumCase(Val, "QualifierKind::Array", flatc::QualifierKind::Array);
+    io.enumCase(Val, "QualifierKind::Const", flatc::QualifierKind::Const);
+  }
+};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::Qualifier>
+  : public TupleLikeMappingTraits<flatc::Qualifier> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::QualifiedType>
+  : public TupleLikeMappingTraits<flatc::QualifiedType> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::EnumEntry>
+  : public TupleLikeMappingTraits<flatc::EnumEntry> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::StructField>
+  : public TupleLikeMappingTraits<flatc::StructField> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::UnionField>
+  : public TupleLikeMappingTraits<flatc::UnionField> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::ArgumentType>
+  : public TupleLikeMappingTraits<flatc::ArgumentType> {};
+
+// Make std::vector<flatc::Qualifier> yaml-serializable as a sequence
+LLVM_YAML_IS_SEQUENCE_VECTOR(flatc::Qualifier)
+
+// Make all types derived from flatc::Type yaml-serializable like tuples
+template<FlatCType T>
+struct llvm::yaml::MappingTraits<T> : public TupleLikeMappingTraits<T> {};
+
+// Make UpcastablePointer<flatc::Type> yaml-serializable polymorphically
+template<>
+struct llvm::yaml::MappingTraits<UpcastablePointer<flatc::Type>>
+  : public PolymorphicMappingTraits<UpcastablePointer<flatc::Type>> {};
+
+template<>
+struct llvm::yaml::MappingTraits<flatc::TypesTree>
+  : public TupleLikeMappingTraits<flatc::TypesTree> {};
```

Comment:
Keep support data structures such as YAML close the definition of the associated struct.

Reply:
Done